tests/yast.test/normalize.test

```diff
@@ -0,0 +1,433 @@
+# 2018-01-08
+#
+# The author disclaims copyright to this source code.  In place of
+# a legal notice, here is a blessing:
+#
+#    May you do good and not evil.
+#    May you find forgiveness for yourself and forgive others.
+#    May you share freely, never taking more than you give.
+#
+#***********************************************************************
+#
+# Tests for the sqlite3_normalize() extension function.
+#
+
+set testdir [file dirname $argv0]
+source $testdir/tester.tcl
+set testprefix normalize
+
+if {![info exists DB]} {set DB ""}
+
+do_test 200 {
+  execsql {
+    CREATE TABLE t1(a,b);
+  }
+} {}
+do_test 201 {
+  set STMT [sqlite3_prepare_v3 $DB \
+      "SELECT a, b FROM t1 WHERE b = ? ORDER BY a;" -1 0 TAIL]
+
+  sqlite3_bind_null $STMT 1
+} {}
+do_test 202 {
+  sqlite3_normalized_sql $STMT
+} {SELECT a,b FROM t1 WHERE b=?ORDER BY a;}
+do_test 203 {
+  sqlite3_finalize $STMT
+} {SQLITE_OK}
+
+do_test 210 {
+  set STMT [sqlite3_prepare_v3 $DB \
+      "SELECT a, b FROM t1 WHERE b = ? ORDER BY a;" -1 2 TAIL]
+
+  sqlite3_bind_null $STMT 1
+} {}
+do_test 211 {
+  sqlite3_normalized_sql $STMT
+} {SELECT a,b FROM t1 WHERE b=?ORDER BY a;}
+do_test 212 {
+  sqlite3_finalize $STMT
+} {SQLITE_OK}
+
+do_test 220 {
+  set STMT [sqlite3_prepare_v3 $DB \
+      "SELECT a, b FROM t1 WHERE b = 'a' ORDER BY a;" -1 2 TAIL]
+} {SELECT a, b FROM t1 WHERE b = 'a' ORDER BY a;}
+do_test 221 {
+  sqlite3_normalized_sql $STMT
+} {SELECT a,b FROM t1 WHERE b=?ORDER BY a;}
+do_test 222 {
+  sqlite3_finalize $STMT
+} {SQLITE_OK}
+
+do_test 297 {
+  execsql {
+    DROP TABLE t1;
+  }
+} {}
+do_test 298 {
+  execsql {
+    CREATE TABLE t1(a,b,c,d,e,col_f,w,x,y,z);
+    CREATE TABLE t2(x,col_y);
+  }
+} {}
+do_test 299 {
+  sqlite3_create_function db
+} {SQLITE_OK}
+
+foreach {tnum sql flags norm} {
+  300
+  {SELECT * FROM t1 WHERE a IN (1) AND b=51.42}
+  0x2
+  {0 {SELECT*FROM t1 WHERE a IN(?,?,?)AND b=?;}}
```

Comment:
Shouldn't we expect one '?' here?

Reply:
No.  I believe the algorithm is designed to normalize all IN clauses to exactly three items.